### PR TITLE
Favicon fix for Windows

### PR DESCRIPTION
**Pull request by abidlabs**
Quick fix so that the favicon path works on windows instead of returning a 404

Closes: #4274

### CODE REVIEW

File: gradio/routes.py [hunk 1]
```diff
@@ -292,7 +292,7 @@ def build_resource(path: str):
         async def favicon():
             blocks = app.get_blocks()
             if blocks.favicon_path is None:
-                return static_resource("img/logo.svg")
+                return static_resource("favicon.svg")
```
Comment (freddyaboulton) [inline gradio/routes.py:295]: Rather than adding a new favicon file - can we use pathlib here to construct the path correctly?

Comment (abidlabs) [reply gradio/routes.py:295]: Yes let's do that, you're right

**Comment by abidlabs** [comment]
Now properly fixed. The issue before was that we were converting all paths to be Windows-style, which meant that any "/" would get converted to "\\" which was seen as insecure by the `safe_join` method. Now, we limit ourselves to the Linux-style separator, as is the case with the original `safe_join` method:

https://github.com/pallets/werkzeug/blob/76b049dd45fd072fb62a54bccc0e8d513b03f4d8/src/werkzeug/security.py#L142

**Review by freddyaboulton — APPROVED** [review]
Thanks @abidlabs !

**Pull request merged**